File: orchestrator/src/test/java/org/cloudifysource/cosmo/config/TestConfig.java
/*******************************************************************************
 * Copyright (c) 2013 GigaSpaces Technologies Ltd. All rights reserved
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *******************************************************************************/

package org.cloudifysource.cosmo.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.support.PropertySourcesPlaceholderConfigurer;
import org.springframework.core.env.ConfigurableEnvironment;
import org.springframework.mock.env.MockPropertySource;

import javax.annotation.PostConstruct;
import javax.inject.Inject;
import java.util.Properties;

/**
 * Abstract test spring configuration.
 *
 * @author Dan Kilman
 * @since 0.1
 */
@Configuration
public class TestConfig {

    @Inject
    private ConfigurableEnvironment environment;

    @PostConstruct
    public void postConstruct() {
        Properties overridenProperties = overridenProperties();
        if (!overridenProperties.isEmpty())
            environment.getPropertySources().addFirst(new MockPropertySource(overridenProperties));
    }

    protected Properties overridenProperties() {
        return new Properties();
    }

    @Bean
    public static PropertySourcesPlaceholderConfigurer stub() {
        return new PropertySourcesPlaceholderConfigurer();
    }
}

File: rest-service/manager_rest/test/test_config.py
########
# Copyright (c) 2014 GigaSpaces Technologies Ltd. All rights reserved
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#        http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
############

import unittest

import manager_rest.config


class TestConfig(unittest.TestCase):

    def test_config_default_amqp_credentials(self):
        config = manager_rest.config.Config()

        self.assertEqual(
            'guest',
            config.amqp_username,
        )
        self.assertEqual(
            'guest',
            config.amqp_password,
        )

    def test_config_default_amqp_ssl_settings(self):
        config = manager_rest.config.Config()

        self.assertEqual(
            False,
            config.amqp_ssl_enabled,
        )
        self.assertEqual(
            '',
            config.amqp_ca_path,
        )

    def test_config_overridden_amqp_credentials(self):
        config = manager_rest.config.Config()

        username = 'newuser'
        password = 'changeme123'

        config.amqp_username = username
        config.amqp_password = password

        self.assertEqual(
            username,
            config.amqp_username,
        )
        self.assertEqual(
            password,
            config.amqp_password,
        )

    def test_config_overridden_amqp_ssl_settings(self):
        config = manager_rest.config.Config()

        cert_path = '/not/a/real/cert.pem'

        config.amqp_ssl_enabled = True
        config.amqp_ca_path = cert_path

        self.assertEqual(
            True,
            config.amqp_ssl_enabled,
        )
        self.assertEqual(
            cert_path,
            config.amqp_ca_path,
        )
